protocols/kad/src/query/peers/closest/disjoint.rs

```diff
@@ -536,12 +536,13 @@ mod tests {
     struct Target(KeyBytes);
 
     impl Arbitrary for Target {
-        fn arbitrary<G: Gen>(g: &mut G) -> Self {
-            Target(Key::from(random_peers(1, g).pop().unwrap()).into())
+        fn arbitrary(_g: &mut Gen) -> Self {
+            Target(Key::from(random_peers(1).pop().unwrap()).into())
         }
     }
 
-    fn random_peers<R: Rng>(n: usize, g: &mut R) -> Vec<PeerId> {
+    fn random_peers(n: usize) -> Vec<PeerId> {
+        let mut g = rand::thread_rng();
```

Comment:
Same as above, can this be done deterministically?

Reply:
Alright, this can be done deterministically, let me play with this a little bit more. I didn't expect this issue to take so long but it looks like I have to choice